### ISSUE TITLE
[Web] onSuggestionTap returns empty searchKey if suggestion is selected via keyboard

### ISSUE THREAD

**Issue by Vovanella95** [maheshmnj/searchfield#138]
Hi!

First of all, very thanks for your beautiful package! We using it in our app and it works great.

I've just found one issue: `onSuggestionTap` returns empty `searchKey` if suggestion is selected via keyboard. It's easy to reproduce on Web platform. 

Could you please fix this issue or let me know, if I'm doing something wrong?

Thanks in advance!

**Video Demo**

https://github.com/maheshmnj/searchfield/assets/11267533/90496d5c-aef5-41a7-90d6-fd4cc973bfb8

**Example code:**
```dart
import 'package:flutter/material.dart';
import 'package:searchfield/searchfield.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _textEditingController = TextEditingController();
  List<String> _suggestions = [];

  @override
  void dispose() {
    _textEditingController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
          SearchField(
            suggestionDirection: SuggestionDirection.flex,
            controller: _textEditingController,
            suggestions: _suggestions
                .map(
                  (e) => SearchFieldListItem(e),
                )
                .toList(),
            onSearchTextChanged: (query) {
              setState(() {
                _suggestions = query.split(' ');
              });
              return null;
            },
            onSuggestionTap: (SearchFieldListItem<dynamic> item) {
              showDialog(
                context: context,
                builder: (_) => Dialog(
                  child: SizedBox(
                    width: 200,
                    height: 120,
                    child:
                        Center(child: Text('Key ${item.searchKey} is clicked')),
                  ),
                ),
              );
            },
            itemHeight: 50,
          ),
          const Expanded(
            child: SizedBox(),
          ),
        ],
      ),
    );
  }
}
```

**Comment by maheshmnj** [comment]
Hi @Vovanella95, Thanks for filing the issue. That definitely looks buggy. Your code looks fine to me.
I will update you on this issue soon.

**Comment by maheshmnj** [comment]
@Vovanella95 Fixed and released in [v1.0.2](https://pub.dev/packages/searchfield/changelog)

**Comment by Vovanella95** [comment]
@maheshmnj  thanks a lot for most quick fix in my life :)
